webpush-subscription-2022.bs

```diff
@@ -0,0 +1,360 @@
+```turtle
+# available at <https://solid.aifb.kit.edu/web-push/service>
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix dc: <http://purl.org/dc/terms/>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix as: <https://www.w3.org/ns/activitystreams#>.
+@prefix ldp: <http://www.w3.org/ns/ldp#>.
+@prefix push: <https://purl.org/solid-web-push/vocab#>.
+            
+<> a as:Profile;
+    rdfs:label "The profile of the Solid Web Push Service"@en;
+    as:summary "The described resource is a Service called Solid Web Push. To subscribe to the service, post the subscription to the provided URI. The subscription must indicate which resource the subscriber whishes to receive updates on.";
+    dc:creator <https://uvdsl.solid.aifb.kit.edu/profile/card#me>;
+    as:describes <#web-push> .
+
+<#web-push> a as:Service;
+    as:name "Solid Web Push"@en;
+    rdfs:label "Solid Web Push Service"@en;
+    ldp:inbox <https://solid.aifb.kit.edu/web-push/subscribe/>; 
+    push:vapidPublicKey "BAOxV1U1Hj5npToInct2VhhYpJkL0GmHqc-ADbHu7O8Z2CJNkqSzc8BfCStWbTKq_yT9B6g6kYjyEHrAEpVuqww"^^xsd:base64Binary.
+```
+Example: Representation of a *subscription service*.
+</div>
+
+Note: The representation includes the subscription endpoint where subscriptions can be submitted via POST using `ldp:inbox`.
+Currently, it is not yet specified by [[!SOLID-NOTIFICATIONS]] how the subscription service is modelled.
+To this end, the example is my suggestion.
+
+
+For Unsubscription,
+
+<div class="advisement"> 
+<b>
+It is not yet specified by [[!SOLID-NOTIFICATIONS]], how unsubscription works. Depends on #36. 
+</b>
+<ul>
+<li>
+Option (1): `as:Undo` via HTTP POST to subscription endpoint.
+</li>
+<li>
+Option (2): HTTP DELETE on a subscription.
+</li>
+</ul>
+</div>
+
+Note: When there is a decision reached, there will be an example included.
+
+
+# Authentication and Authorization # {#auth}
+
+As described by the Solid Notifications Protocol section on Authorization,
+the WebPush subscription API requires authorization and follows the guidance of the Solid Protocol
+sections on Authentication and Authorization [[!SOLID-PROTOCOL]].
+
+It is beyond the scope of this document to describe how a client fetches an access token.
+Solid-OIDC is one example of an authentication mechanism that could be used with Solid Notifications [[!SOLID-OIDC]].
+
+
+<pre class=biblio>
+{
+    "SOLID-PROTOCOL": {
+        "authors": [
+            "Sarven Capadisli",
+            "Tim Berners-Lee",
+            "Ruben Verborgh",
+            "Kjetil Kjernsmo"
+        ],
+        "href": "https://solidproject.org/TR/protocol",
+        "title": "Solid Protocol",
+        "publisher": "W3C Solid Community Group"
+    },
+    "SOLID-NOTIFICATIONS": {
+        "authors": [
+            "Aaron Coburn",
+            "Sarven Capadisli"
+        ],
+        "href": "https://solid.github.io/notifications/protocol",
```

Comment:
not to change now, but just FYI that this will (soon?) be under the `https://solidproject/TR/` URI space